src/phoenix/logging/_config.py

```diff
@@ -0,0 +1,90 @@
+import atexit
+import logging
+import logging.config
+import logging.handlers
+import queue
+from sys import stderr, stdout
+
+# from typing import override
+from phoenix.config import LoggingMode
+from phoenix.logging._filter import NonErrorFilter
+from phoenix.settings import Settings
+
+from ._formatter import PhoenixJSONFormatter
+
+
+def setup_logging() -> None:
+    """
+    Configures logging for the specified logging mode.
+    """
+    logging_mode = Settings.logging_mode
+    if logging_mode is LoggingMode.AS_LIBRARY:
+        _setup_library_logging()
+    elif logging_mode is LoggingMode.AS_APPLICATION:
+        _setup_application_logging()
+    else:
+        raise ValueError(f"Unsupported logging mode: {logging_mode}")
+
+
+def _setup_library_logging() -> None:
+    """
+    Configures logging if Phoenix is used as a library
+    """
+    logger = logging.getLogger("phoenix")
+    logger.setLevel(Settings.logging_level)
+    db_logger = logging.getLogger("sqlalchemy")
+    db_logger.setLevel(Settings.db_logging_level)
+    logger.info("Default logging ready")
+
+
+def _setup_application_logging() -> None:
+    """
+    Configures logging if Phoenix is used as an application
+    """
+    root_logger = logging.getLogger()
+    # Remove all existing handlers
+    for handler in root_logger.handlers[:]:
+        root_logger.removeHandler(handler)
+        handler.close()
+
+    sql_engine_logger = logging.getLogger("sqlalchemy.engine.Engine")
+    # Remove all existing handlers
+    for handler in sql_engine_logger.handlers[:]:
+        sql_engine_logger.removeHandler(handler)
+        handler.close()
+
+    phoenix_logger = logging.getLogger("phoenix")
+    phoenix_logger.setLevel(Settings.logging_level)
+    sql_logger = logging.getLogger("sqlalchemy")
+    sql_logger.setLevel(Settings.db_logging_level)
+
+    fmt_keys = {
+        "level": "levelname",
+        "message": "message",
+        "timestamp": "timestamp",
+        "logger": "name",
+        "module": "module",
+        "function": "funcName",
+        "line": "lineno",
+        "thread_name": "threadName",
+    }
+    formatter = PhoenixJSONFormatter(fmt_keys=fmt_keys)
+
+    stdout_handler = logging.StreamHandler(stdout)
+    stdout_handler.setFormatter(formatter)
+    stdout_handler.setLevel(Settings.logging_level)
+    stdout_handler.addFilter(NonErrorFilter())
```

Comment:
Is this to ensure errors are going to standard error?

Reply:
Correct